tests/integration/telemetry/test_telemetry_contract.py

```diff
@@ -0,0 +1,73 @@
+
+from .integ_base import IntegBase, TelemetryServer
+
+
+class TestTelemetryContract(IntegBase):
+    """
+    Validates the basic tenets/contract Telemetry module needs to adhere to
+    """
+
+    def test_must_not_send_metrics_if_disabled_using_envvar(self):
+        """
+        No metrics should be sent if "Enabled via Config file but Disabled via Envvar"
+        """
+        # Enable it via configuration file
+        self.set_config(telemetry_enabled=True)
```

Comment:
Personally, I am not crazy about setting this config through code, as these are integ tests and should be set how customers would enable/disable. I am ok with it, just would prefer always going through customer paths in our integ tests.

Reply:
This is how customers would enable telemetry. Thru the config.

Reply:
There is a sight nuance to me though. This function gets called through command executions for the customers. The customer would never explicitly call this method. I know it’s what happens for them but since a customer never directly interacts with it, calling the method in an integ test looks off to me. Not blocking on this, just something I noticed

Reply:
Sure, but we need some way of mimicing customer's environment. Calling this method is a way to do that